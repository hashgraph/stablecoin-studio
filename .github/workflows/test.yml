name: test
on:
  pull_request:
    branches: [ main, release/**, feature/** ]
  push:
    branches: [ main, release/** ]
    tags: [ v* ]
jobs:
  test-node:
    name: Unit Tests
    runs-on: ubuntu-latest
    steps:
      - name: Use Node.js TLS 16.x
        uses: actions/setup-node@v1
        with:
          node-version: 16.x
      - name: Checkout repository
        uses: actions/checkout@v2
      - name: Install dependencies
        run: npm i
      - name: Build
        run: npm run build
      - name: Test
        run: npm run test