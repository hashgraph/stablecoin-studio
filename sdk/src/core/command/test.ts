/* eslint-disable @typescript-eslint/no-empty-interface */

interface CommandResponse {}
interface ICommand {}

export class Command<T> implements ICommand {
	private $resultType!: T;
}

export interface ICommandBus {
	execute<X>(command: Command<X>): Promise<X>;
}

export type IInferringCommandHandler<CommandType extends Command<unknown>> =
	CommandType extends Command<infer ResultType>
		? {
				execute(command: CommandType): Promise<ResultType>;
		  }
		: never;

class ConcreteCommandResponse implements CommandResponse {
	constructor(public readonly status: string) {}
}

class ConcreteCommand extends Command<ConcreteCommandResponse> {
	constructor(public readonly value: string) {
		super();
	}
}

class CommandHandler<CommandType extends Command<unknown>>
	implements IInferringCommandHandler<ConcreteCommand>
{
	execute(command: ConcreteCommand): Promise<ConcreteCommandResponse> {
		throw new Error('Method not implemented.');
	}
}

class CommandBus implements ICommandBus {
	execute<X>(command: Command<X>): Promise<X> {
		throw new Error('Method not implemented.');
	}
}

// Test

const bus = new CommandBus();
const res = bus.execute(new ConcreteCommand('1'));
