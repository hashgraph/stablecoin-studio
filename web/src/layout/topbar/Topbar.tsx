import { Box, Button, Flex, Image, Text, useMediaQuery } from '@chakra-ui/react';
import { useTranslation } from 'react-i18next';
import { useNavigate } from 'react-router-dom';
import LOGO_HEDERA from '../../assets/svg/hedera-hbar-logo.svg';
import Icon from '../../components/Icon';
import { NamedRoutes } from '../../Router/NamedRoutes';
import { RouterManager } from '../../Router/RouterManager';
import CoinDropdown from './CoinDropdown';
import TopbarRight from './TopbarRight';

const Topbar = () => {
	const { t } = useTranslation('global');
	const navigate = useNavigate();
	const [isLargerThan1210] = useMediaQuery('(min-width: 1210px)');

	const handleNavigateSC = () => {
		RouterManager.to(navigate, NamedRoutes.StableCoinCreation);
	};

	return (
		<Flex
			data-testid='topbar'
			w='100%'
			h='64px'
			boxShadow='down-black'
			bgColor='brand.white'
			color='brand.gray2'
			alignItems='center'
			position='relative'
			zIndex='100'
		>
			<Box minW='80px' w='80px' textAlign='center' ml='64px' mr='64px'>
				<Image data-testid='topbar-logo' src={LOGO_HEDERA} w='40px' h='40px' margin='auto' />
			</Box>
			<Box borderRight='2px solid' borderRightColor='light.primary' w='1px' h='30px' />
			<Flex w='100%' h='100%' justifyContent='space-between' alignItems='center' pl={6} pr={10}>
				<Flex gap={5}>
					<CoinDropdown />
					<Button
						variant='secondary'
						onClick={handleNavigateSC}
						gap={2}
						w={isLargerThan1210 ? '182px' : '30px'}
					>
						<Icon name='Plus' fontSize='16px' />
						{isLargerThan1210 && (
							<Text fontSize='14px' fontWeight='400'>
								{t('topbar.createSC')}
							</Text>
						)}
					</Button>
				</Flex>
				<TopbarRight />
			</Flex>
		</Flex>
	);
};

export default Topbar;
