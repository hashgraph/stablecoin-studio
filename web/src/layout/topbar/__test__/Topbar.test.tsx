import Topbar from '../Topbar';
import { render } from '../../../test/';

const HEDERA_LOGO = 'hedera-hbar-logo.svg';

describe(`<${Topbar.name} />`, () => {
	test('should render correctly', () => {
		const component = render(<Topbar />);

		const topbar = component.getByTestId('topbar');
		expect(topbar).toBeInTheDocument();
		expect(component.asFragment()).toMatchSnapshot();
	});

	test('should has Hedera Logo', () => {
		const component = render(<Topbar />);

		const logo = component.getByTestId('topbar-logo');
		expect(logo).toBeInTheDocument();
		expect(logo).toHaveAttribute('src', HEDERA_LOGO);
	});

	test('should has a component in left position', () => {
		const component = render(<Topbar />);

		const topbarLeft = component.getByTestId('topbar-left');
		expect(topbarLeft).toBeInTheDocument();
	});

	test('should has a component in right position', () => {
		const component = render(<Topbar />);

		const TopbarRight = component.getByTestId('topbar-right');
		expect(TopbarRight).toBeInTheDocument();
	});
});
