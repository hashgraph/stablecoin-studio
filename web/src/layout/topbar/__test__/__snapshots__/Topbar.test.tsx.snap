// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Topbar /> should render correctly 1`] = `
<DocumentFragment>
  <div
    class="css-1njzpr3"
    data-testid="topbar"
  >
    <div
      class="css-gut43g"
    >
      <img
        class="chakra-image css-z4bd0k"
        data-testid="topbar-logo"
        src="hedera-hbar-logo.svg"
      />
    </div>
    <div
      class="css-1grq7v1"
    />
    <div
      class="css-1xcxk9t"
    >
      <div
        class="css-1lbjv5i"
      >
        <div
          class="css-1rc4l3d"
          data-testid="coin-dropdown"
        >
          <div
            class="chakra-stack css-1h0sumw"
          >
            <div
              class="chakra-form-control css-1kxonj9"
              data-testid="form_control"
              role="group"
            >
              <div
                class="css-frrdvf"
              >
                <span
                  class="css-1f43avz-a11yText-A11yText"
                  id="react-select-2-live-region"
                />
                <span
                  aria-atomic="false"
                  aria-live="polite"
                  aria-relevant="additions text"
                  class="css-1f43avz-a11yText-A11yText"
                />
                <div
                  class=" css-lub9qs"
                >
                  <div
                    class="css-3y2vuh"
                  >
                    <span
                      class="chakra-text css-4pvr19"
                      data-testid="select-placeholder"
                    >
                      0.0.49319786 - TEST
                    </span>
                    <div
                      class=" css-svnjlw"
                      data-value=""
                    >
                      <input
                        aria-autocomplete="list"
                        aria-describedby="react-select-2-placeholder"
                        aria-expanded="false"
                        aria-haspopup="true"
                        autocapitalize="none"
                        autocomplete="off"
                        autocorrect="off"
                        class=" css-146k1ar-input"
                        id="field-:r1:"
                        role="combobox"
                        spellcheck="false"
                        tabindex="0"
                        type="text"
                        value=""
                      />
                    </div>
                  </div>
                  <div
                    class=" css-hfbj6y"
                  >
                    <div
                      class="css-1aabyhj"
                    >
                      <svg
                        class="chakra-icon css-1r5bg7h"
                        fill="currentColor"
                        focusable="false"
                        height="1em"
                        viewBox="0 0 256 256"
                        width="1em"
                        xmlns="http://www.w3.org/2000/svg"
                      >
                        <rect
                          fill="none"
                          height="256"
                          width="256"
                        />
                        <polyline
                          fill="none"
                          points="208 96 128 176 48 96"
                          stroke="currentColor"
                          stroke-linecap="round"
                          stroke-linejoin="round"
                          stroke-width="16"
                        />
                      </svg>
                    </div>
                  </div>
                </div>
                <input
                  name="coin-dropdown"
                  type="hidden"
                  value=""
                />
              </div>
            </div>
          </div>
        </div>
        <div
          class="chakra-stack css-84zodg"
        >
          <div
            class="css-0"
          >
            <button
              class="chakra-button css-r4h8vy"
              type="button"
            >
              <svg
                class="chakra-icon css-1ita9rw"
                fill="currentColor"
                focusable="false"
                height="1em"
                viewBox="0 0 256 256"
                width="1em"
                xmlns="http://www.w3.org/2000/svg"
              >
                <rect
                  fill="none"
                  height="256"
                  width="256"
                />
                <line
                  fill="none"
                  stroke="currentColor"
                  stroke-linecap="round"
                  stroke-linejoin="round"
                  stroke-width="16"
                  x1="40"
                  x2="216"
                  y1="128"
                  y2="128"
                />
                <line
                  fill="none"
                  stroke="currentColor"
                  stroke-linecap="round"
                  stroke-linejoin="round"
                  stroke-width="16"
                  x1="128"
                  x2="128"
                  y1="40"
                  y2="216"
                />
              </svg>
            </button>
          </div>
        </div>
      </div>
      <div
        class="css-166re2f"
        data-testid="topbar-right"
      >
        <div
          class="chakra-stack css-d17m5h"
        >
          <div
            class="chakra-stack css-jp20jv"
          >
            <div
              class="css-1m8iww1"
            >
              <p
                class="chakra-text css-1oyarqz"
                data-testid="topbar-right-account"
              />
            </div>
            <div
              class="css-1m8iww1"
            >
              <p
                class="chakra-text css-gegx7f"
                data-testid="topbar-right-network"
              />
            </div>
          </div>
          <svg
            fill="#fdfdfc"
            height="22"
            viewBox="0 0 256 256"
            width="22"
            xmlns="http://www.w3.org/2000/svg"
          >
            <rect
              fill="none"
              height="256"
              width="256"
            />
            <circle
              cx="128"
              cy="128"
              fill="none"
              r="96"
              stroke="#fdfdfc"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-width="12"
            />
            <circle
              cx="128"
              cy="180"
              r="10"
            />
            <path
              d="M128,144v-8a28,28,0,1,0-28-28"
              fill="none"
              stroke="#fdfdfc"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-width="12"
            />
          </svg>
        </div>
        <div
          class="css-dv2lk7"
        />
        <div
          class="css-hf30mn"
        >
          <svg
            class="chakra-icon css-dee54c"
            fill="currentColor"
            focusable="false"
            height="1em"
            viewBox="0 0 256 256"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <rect
              fill="none"
              height="256"
              width="256"
            />
            <line
              fill="none"
              stroke="currentColor"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-width="16"
              x1="128"
              x2="128"
              y1="48"
              y2="124"
            />
            <path
              d="M176,54.2a88,88,0,1,1-96,0"
              fill="none"
              stroke="currentColor"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-width="16"
            />
          </svg>
        </div>
      </div>
    </div>
  </div>
  <span
    hidden=""
    id="__chakra_env"
  />
</DocumentFragment>
`;
